rewindable-index/src/RewindableIndex/Storable.hs

```diff
@@ -267,8 +318,8 @@ flushBuffer s = do
       mx = s ^. config . memoryBufferSize
   if mx == cr
   then do
-    v  <- V.freeze es
-    h' <- persistToStorage v (s ^. handle)
+    es' <- foldEvents <$> V.freeze es
+    h' <- persistToStorage es' (s ^. handle)
```

Comment:
I know it's a breaking change but don't we want to also allow developer to persist synthetic event if they want to, just to track on disk how far the indexer went?

Reply:
I think we do. I am not sure about it though so I went for the least disruptive change.